### ISSUE TITLE
Relabelling proposals

### ISSUE THREAD

**Issue by benjub** [metamath/set.mm#840]
* For consistency, it looks like imp3a and exp3a could be relabelled impd and expd respectively.

* When axioms have variants (e.g. inference form, variant with dv condition, with existential quantifier...), some labels "lose their 'x'" and some do not, for instance: a1i, a1d, a6ev, a6e... versus ax5e, ax6v, ax12i.  Is there a rule behind that, or should we try to relabel them more consistently ?

**Comment by nmegill** [comment]
> * For consistency, it looks like imp3a and exp3a could be relabelled impd and expd respectively.

imp3a is part of a series imp3a, imp4a, imp4b,... in which not all of the implications become conjunctions, unlike impnn (imp31, imp32, etc.).  However, renaming it to impd is consistent with the *d naming and is probably easier to remember.  (Same with expd.)  So go ahead and rename it.

> * When axioms have variants (e.g. inference form, variant with dv condition, with existential quantifier...), some labels "lose their 'x'" and some do not, for instance: a1i, a1d, a6ev, a6e... versus ax5e, ax6v, ax12i.  Is there a rule behind that, or should we try to relabel them more consistently ?

There is no rule.  We should probably use x in all of them.  You can go ahead and do that if you want.  Make sure no naming conflicts arise of course.  And document them in the label change list.

While you are doing this, you might also check that the ones without the x correspond to the new names at http://us.metamath.org/mpeuni/mmset.html#oldaxioms in case I missed one during my renaming in December.

There is a small potential problem:  if you have both ax-n and axn, 'verify markup' will complain if they don't have exactly the same content, because axn is typically used to prove ax-n from alternate axioms.  If that occurs you'll have to use a different name for axn.

**Comment by benjub** [comment]
I'll do this progressively.  Actually, my original thought was to drop the "x" everywhere (I'm restricting myself to the "numbered" axioms), given that a1i a1d... are pretty common, and also because I think this would create fewer conflicts: labels beginning by "a" followed by a digit are very rare. I think that ax1 ax2... should be reserved for when the statement is exactly the same as that of ax-1 ax-2...

So basically, I would remove the "x" from ax5e ax6v ax12i...

**Comment by nmegill** [comment]
> Actually, my original thought was to drop the "x" everywhere...I think that ax1 ax2... should be reserved for when the statement is exactly the same as that of ax-1 ax-2...

OK, I'll accept that.  You can do ahead and do that for ax-1 to ax-13 at your convenience as far as I'm concerned.

One problem is that the old axioms axc4, axc7,... would become ac4, ac7,... which normally designates the Axiom of Choice.  Actually ac4 and ac7 are already there as versions of AC.  Perhaps let axc4, axc7,... continue to use ax.

Feel free to post a newsgroup message if you want other opinions (not everyone reads the github messages),

**Comment by benjub** [comment]
I think you are right: it's probably better to adopt ax* as a rule.  There would be far fewer changes.  The only affected labels would be:
-------
66 a1i $p
120 a1d $p
128 a1ii $p
132 a1iiALT $p
219 a1dd $p
1248 a1bi $p
3729 a1tru $p
112034 a1i13 $p
112038 a1i4 $p
112042 a1i14 $p
112046 a1i24 $p
112050 a1i34 $p
-------
75 a2i $p
124 a2d $p
-------
4336 a5d $p
-------
4500 a6ev $p
5316 a6e $p
125895 a6e2eq $p
125902 a6e2nd $p
125909 a6e2ndeq $p
127862 a6e2eqVD $p
127869 a6e2ndVD $p
127876 a6e2ndeqVD $p
127979 a6e2ndALT $p
127986 a6e2ndeqALT $p
132693 a6er $p
-------
4755 a10e $p
-------
5608 a16nf $p
5612 a16gb $p
5613 a16nfALT $p
5876 a16gALT $p
6236 a16g-o $p
-------
The statements a1i, a1d, a2d are used a lot, so maybe it would be nice to keep them short.  So a rule could be:
* For statements derived from (but not identical to) propcalc axioms, we keep a*, and for statements derived from the other axioms, we use ax*.
Therefore, the only changes would be for the 18 above statements derived from ax-5,6,10,16.

**Comment by nmegill** [comment]
That is OK with me.
